src/app/reporting/ReportScheduler.h

```diff
@@ -167,6 +167,18 @@ class ReportScheduler : public ReadHandler::Observer
 
     virtual void ReportTimerCallback() = 0;
 
+    void OnActiveModeEntered()
+    {
+        for (auto & iter : mReadHandlerList)
+        {
+            if (iter.IsReportableNow())
+            {
+                OnBecameReportable(iter.GetReadHandler());
```

Comment:
Right, this doesn't guarantee report emission, the logic needs a rework because on one hand we do not want to force dirty on readHandlers that are not immediately reportable. 

> Is there a reason this is implemented here and not in ReportSchedulerImpl? Will all report schedulers want this behavior?

This is in ReportScheduler so it can be accessed from a pointer to a report Scheduler, regardless of the implementation. The ICD manager aims to leverage synchronised subs when using this but it should work regardless of the implementation.

> The naming is confusing and at least needs documentation.
Documentation added, name changed.

> The going from node to handler and then back to mode we end up with is a bit silly, but might be ok to share more code.
Replaced by ForceDirtyState since otherwise scheduleReport would not schedule an immediate report.

Reply:
> This is in ReportScheduler so it can be accessed from a pointer to a report Scheduler

Yes, but why is it not pure virtual here?